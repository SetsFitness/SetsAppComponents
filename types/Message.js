/**
 * TODO
 */
type Message = {
  id: string,
  board: string,
  item_type: string,
  marker: number,
  time_created: string,
  from: string,
  type: string,
  name: string,
  profileImagePath: string,
  message: string,
  lastSeenFor: [string],
}
export default Message;
