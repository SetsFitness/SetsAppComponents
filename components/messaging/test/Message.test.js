import React from 'react';
import { shallow } from 'enzyme';
import TestConfig, {store} from "../../../testing/TestConfig";
import Message from "../Message";

TestConfig();
it('renders without crashing', () => {
    const component = shallow(<Message userID={null} message={null} store={store()}/>);
    expect(component).toMatchSnapshot();
});
