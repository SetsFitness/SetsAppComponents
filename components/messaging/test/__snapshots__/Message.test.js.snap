// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    message={
      Object {
        "from": "CL0001",
        "message": "Hey there Blinky",
        "messageURL": "amazonaws.com",
        "name": "Sleeny",
        "profilePicture": "this/is/a/fake/path",
        "type": "picture",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    userID="CL0001"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Button
            as="button"
            floated="right"
            labelPosition="right"
            role="button"
          >
            <Label
              className="ui right fluid"
              color="purple"
              pointing="right"
            >
              <div
                className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                style={
                  Object {
                    "backgroundImage": "url(amazonaws.com)",
                  }
                }
              />
            </Label>
          </Button>
        </div>,
        <br />,
        <br />,
        <br />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            as="button"
            floated="right"
            labelPosition="right"
            role="button"
          >
            <Label
              className="ui right fluid"
              color="purple"
              pointing="right"
            >
              <div
                className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                style={
                  Object {
                    "backgroundImage": "url(amazonaws.com)",
                  }
                }
              />
            </Label>
          </Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": <Label
              className="ui right fluid"
              color="purple"
              pointing="right"
            >
              <div
                className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                style={
                  Object {
                    "backgroundImage": "url(amazonaws.com)",
                  }
                }
              />
            </Label>,
            "floated": "right",
            "labelPosition": "right",
            "role": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                style={
                  Object {
                    "backgroundImage": "url(amazonaws.com)",
                  }
                }
              />,
              "className": "ui right fluid",
              "color": "purple",
              "pointing": "right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4",
                "style": Object {
                  "backgroundImage": "url(amazonaws.com)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Button
              as="button"
              floated="right"
              labelPosition="right"
              role="button"
            >
              <Label
                className="ui right fluid"
                color="purple"
                pointing="right"
              >
                <div
                  className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                  style={
                    Object {
                      "backgroundImage": "url(amazonaws.com)",
                    }
                  }
                />
              </Label>
            </Button>
          </div>,
          <br />,
          <br />,
          <br />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              as="button"
              floated="right"
              labelPosition="right"
              role="button"
            >
              <Label
                className="ui right fluid"
                color="purple"
                pointing="right"
              >
                <div
                  className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                  style={
                    Object {
                      "backgroundImage": "url(amazonaws.com)",
                    }
                  }
                />
              </Label>
            </Button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "children": <Label
                className="ui right fluid"
                color="purple"
                pointing="right"
              >
                <div
                  className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                  style={
                    Object {
                      "backgroundImage": "url(amazonaws.com)",
                    }
                  }
                />
              </Label>,
              "floated": "right",
              "labelPosition": "right",
              "role": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4"
                  style={
                    Object {
                      "backgroundImage": "url(amazonaws.com)",
                    }
                  }
                />,
                "className": "ui right fluid",
                "color": "purple",
                "pointing": "right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "u-avatar u-avatar--large u-margin-x--auto u-margin-top--neg4",
                  "style": Object {
                    "backgroundImage": "url(amazonaws.com)",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
